pkg/cmd/get.go

```diff
@@ -0,0 +1,182 @@
+				return err
+			}
+
+			err = o.Complete(args)
+			if err != nil {
+				return err
+			}
+
+			err = o.Run()
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	o.kubeFlags.AddFlags(cmd.Flags())
+	cmd.Flags().BoolVarP(&o.AllNamespaces, "all-namespaces", "A", o.AllNamespaces, "If present, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
+	cmd.Flags().StringVar(&o.KubeArchiveHost, "kubearchive-host", o.KubeArchiveHost, fmt.Sprintf("Host where the KubeArchive API Server is listening. Defaults to '%s'", o.KubeArchiveHost))
+
+	return cmd
+}
+
+func (o *GetOptions) Validate(args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("You must specify the type of resource to get.")
+	}
+
+	if len(args) > 1 {
+		return fmt.Errorf("You must specify just one resource to get.")
+	}
+
+	return nil
+}
+
+func (o *GetOptions) Complete(args []string) error {
+	o.Resource = args[0]
+	if strings.HasPrefix(o.Resource, "v1") {
+		o.IsCoreResource = true
+	}
+
+	o.APIPath = fmt.Sprintf("/apis/%s", o.Resource)
+	if o.IsCoreResource {
+		o.APIPath = fmt.Sprintf("/api/%s", o.Resource)
+	}
+
+	config, err := o.kubeFlags.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	o.RESTConfig = config
+
+	return nil
+}
+
+// TODO: when our API returns a List resource, we can remove the decode function
+func (o *GetOptions) getResources(host string, decode func([]byte) ([]unstructured.Unstructured, error)) ([]unstructured.Unstructured, error) {
+	url := fmt.Sprintf("%s%s", host, o.APIPath)
+
+	client, err := rest.HTTPClientFor(o.RESTConfig)
+	if err != nil {
+		return nil, err
+	}
+	response, err := client.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
+		bBytes, err := io.ReadAll(response.Body)
+		if err != nil {
+			return nil, err
+		}
+		return nil, fmt.Errorf("Status code != 200, but %d with body %s", response.StatusCode, string(bBytes))
+	}
+
+	bodyBytes, err := io.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return decode(bodyBytes)
+}
+
+func (o *GetOptions) getClusterResources() ([]unstructured.Unstructured, error) {
+	// TODO: when our API returns a List resource, we can remove the decode function
+	return o.getResources(o.RESTConfig.Host, func(bodyBytes []byte) ([]unstructured.Unstructured, error) {
+		var list unstructured.UnstructuredList
+		err := json.Unmarshal(bodyBytes, &list)
+		if err != nil {
+			return nil, err
+		}
+		return list.Items, nil
+	})
+}
+
+func (o *GetOptions) getKubearchiveResources() ([]unstructured.Unstructured, error) {
+	o.RESTConfig.CAData = nil      // Remove CA data from the Kubeconfig
+	o.RESTConfig.CAFile = "ca.crt" // This expects you to have extracted the CA, see DEVELOPMENT.md
+	o.RESTConfig.BearerToken = *o.kubeFlags.BearerToken
+
+	// TODO: when our API returns a List resource, we can remove the decode function
```

Comment:
I don't see a `decode` function in the code below this comment. Is this a leftover TODO that has already been done?

Reply:
I changed these comments slightly, the `decode` function is the inlined function below. It gets passed as a parameter called `decode` to `o.getResources` because it decodes the body into a certain type. Its needed because our API returns a list of resources, and the K8s API returns a resource `List`.